app/src/main/java/at/aau/serg/activities/GameScreenActivity.kt

```diff
@@ -56,28 +59,36 @@ class GameScreenActivity : AppCompatActivity() {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             initialCards = intent.getSerializableExtra("cards", Array<CardItem>::class.java)
             initialTrumpCard = intent.getSerializableExtra("trump", CardItem::class.java)
+            myPlayerIndex = intent.getSerializableExtra("me", Int::class.java)!!
         } else {
             initialCards = intent.getSerializableExtra("cards") as? Array<CardItem>
             initialTrumpCard = intent.getSerializableExtra("trump") as? CardItem
+            myPlayerIndex = (intent.getSerializableExtra("me") as? Int)!!
```

Comment:
If the item is an int you can use `getIntExtra` instead of `getSerializableExtra` (see `playerCount` for an example)